packages/block-library/src/table-of-contents/edit.js

```diff
@@ -145,6 +148,42 @@ export default function TableOfContentsEdit( {
 						}
 					/>
 				</ToolsPanelItem>
+				<ToolsPanelItem
+					hasValue={ () => maxLevel }
```

Comment:
```suggestion
					hasValue={ () => !! maxLevel }
```

This needs to be a boolean; otherwise, the tools panel item won't be displayed.